lightning/src/routing/router.rs

```diff
@@ -876,39 +877,85 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, paye
+				let mut aggregate_next_hops_fee_msat: u64 = 0;
+				let mut aggregate_next_hops_path_htlc_minimum_msat: u64 = 0;
+
+				for (idx, hop) in route_it.enumerate() {
+					// BOLT 11 doesn't allow inclusion of features for the last hop hints, which
+					// really sucks, cause we're gonna need that eventually.
+					let hop_htlc_minimum_msat: u64 = hop.htlc_minimum_msat.unwrap_or(0);
+
+					let directional_info = DummyDirectionalChannelInfo {
+						cltv_expiry_delta: hop.cltv_expiry_delta as u32,
+						htlc_minimum_msat: hop_htlc_minimum_msat,
+						htlc_maximum_msat: hop.htlc_maximum_msat,
+						fees: hop.fees,
+					};
+
+					let prev_hop = if idx == 0 { &payee } else { &rev_route[idx-1].src_node_id };
+
+					let reqd_channel_cap = Some(final_value_msat.checked_add(match idx {
+						0 => 999,
+						_ => aggregate_next_hops_fee_msat.checked_add(999).unwrap_or(u64::max_value())
+					}).unwrap_or(u64::max_value()) / 1000); // converting from msat
+
+
+					// We assume that the recipient only included route hints for routes which had
+					// sufficient value to route `final_value_msat`. Note that in the case of "0-value"
+					// invoices where the invoice does not specify value this may not be the case, but
+					// better to include the hints than not.
+					if !add_entry!(hop.short_channel_id, hop.src_node_id, prev_hop, directional_info, reqd_channel_cap, &empty_channel_features, aggregate_next_hops_fee_msat, path_value_msat, aggregate_next_hops_path_htlc_minimum_msat) {
+						// If this hop was not used then there is no use checking the preceding hops
+						// in the RouteHint. We can break by just searching for a direct channel between
+						// last checked hop and first_hop_targets
+						hop_used = false;
+					}
+
+					// Searching for a direct channel between last checked hop and first_hop_targets
+					if let Some(&(ref first_hop, ref features, ref outbound_capacity_msat, _)) = first_hop_targets.get(&prev_hop) {
+						add_entry!(first_hop, *our_node_id , prev_hop, dummy_directional_info, Some(outbound_capacity_msat / 1000), features, aggregate_next_hops_fee_msat, path_value_msat, aggregate_next_hops_path_htlc_minimum_msat);
+					}
+
+					if !hop_used {
+						break;
+					}
+
+					// In the next values of the iterator, the aggregate fees already reflects
+					// the sum of value sent from payer (final_value_msat) and routing fees
+					// for the last node in the RouteHint. We need to just add the fees to
+					// route through the current node so that the preceeding node (next iteration)
+					// can use it.
+					aggregate_next_hops_fee_msat = aggregate_next_hops_fee_msat.checked_add(compute_fees(aggregate_next_hops_fee_msat + final_value_msat, hop.fees).unwrap_or(0)).unwrap_or(u64::max_value());
+
+					aggregate_next_hops_path_htlc_minimum_msat = aggregate_next_hops_path_htlc_minimum_msat.checked_add(compute_fees(hop_htlc_minimum_msat, hop.fees).unwrap_or(0)).unwrap_or(u64::max_value());
```

Comment:
We need to calculate fees on the current backwards-walking path minimum, but `cmp::max` it with the current hop minimum, like this, I think:

```suggestion
					aggregate_next_hops_path_htlc_minimum_msat = cmp::max(hop_htlc_minimum_msat,
						aggregate_next_hops_path_htlc_minimum_msat.checked_add(
							compute_fees(aggregate_next_hops_path_htlc_minimum_msat, hop.fees).unwrap_or(u64::max_value())
						).unwrap_or(u64::max_value()));
```

Reply:
I was thinking that since this kind of comparison was already being done in the `add_entry` macro, it was not needed at this point.
But yes, you are right, we need to find the maximum of the two to satisfy to consider routing through the next hops. Committing the changes.